src/pybind/mgr/dashboard/frontend/src/app/core/error/error.component.ts

```diff
@@ -0,0 +1,45 @@
+import { Location } from '@angular/common';
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'cd-error',
+  templateUrl: './error.component.html',
+  styleUrls: ['./error.component.scss']
+})
+export class ErrorComponent implements OnInit {
+  topHeader: string;
+  topMessage: string;
+  btmHeader: string;
+  btmMessage: string;
+  section: string;
+  section_info: string;
+  fa_icon: string;
+
+  constructor(public route: ActivatedRoute, private location: Location) {}
+
+  ngOnInit() {
+    try {
+      this.topMessage = history.state.topMessage;
+      this.topHeader = history.state.topHeader;
+      this.btmHeader = history.state.btmHeader;
+      this.btmMessage = history.state.btmMessage;
+      this.section = history.state.section;
+      this.section_info = history.state.section_info;
+      this.fa_icon = history.state.fa_icon;
+
+      if (this.topMessage === '' || this.topMessage === undefined) {
+        this.topHeader = $localize`Client Error`;
+        this.topMessage = $localize`Sorry for the inconvinience, we have encountered a Client Error.`;
+        this.btmHeader = $localize`Page Not Found`;
+        this.btmMessage = $localize`Sorry, we couldn’t find what you were looking for.
+            The page you requested may have been changed or moved.`;
+        this.fa_icon = 'fa fa-exclamation-circle';
+      }
+    } catch (error) {}
```

Comment:
What's this `try-catch` for? I mean, we are ignoring the exceptions, right? Is that what we want here?

Reply:
By not covering this code under try-catch block, there was some continuous error in the make check jenkins job regarding 'Uncaught promise rejection unhandled', therefore decided to have this code under the try-catch block.

Reply:
Yeah, I'd at least try to print some info about that. It's not a good thing to catch general exceptions and then ignoring them :S

Reply:
@Devp00l , I think there is an assync call involved in using history.state object, therefore it needs to be wrapped inside the try-catch block in case there is an exception.

Reply:
Sure, @epuertat , will print the error message in console.